taxi/src/lib.rs

```diff
@@ -89,24 +97,40 @@ impl Game {
 		game
 	}
 
+	/// Returns size of the game world
+	pub fn world_size(&self) -> (usize, usize) {
+		(Self::WORLD_HEIGHT, Self::WORLD_WIDTH)
+	}
+
+	/// Returns true if player has won the game
 	pub fn has_won(&self) -> bool {
 		self.position == self.goal && self.picked_up
 	}
 
+	/// Returns player position
+	pub fn player_position(&self) -> (u32, u32) {
+		self.position
+	}
+
+	/// Returns true if passenger has been picked up
+	pub fn passenger_picked_up(&self) -> bool {
+		self.picked_up
+	}
+
+	/// Returns a distance to a game goal
 	pub fn distance_to_goal(&self) -> f64 {
-		((((self.goal.0 as i32 - self.position.0 as i32).pow(2) + (self.goal.1 as i32 - self.position.1 as i32).pow(2)) as f64).sqrt())
+		distance(self.goal, self.position)
 	}
 
+	/// Returns a distance to a passenger
 	pub fn distance_to_passenger(&self) -> f64 {
-		((((self.passenger.0 as i32 - self.position.0 as i32).pow(2)
-		+ (self.passenger.1 as i32 - self.position.1 as i32).pow(2)
-		) as f64).sqrt())
+		distance(self.passenger, self.position)
 	}
 
 	/// Enter move directly instead of starting an stdin loop, for instance from an automated player
 	/// Returns a bool signifying whether or not this move lead to winning the game
-	pub fn enter_move(&mut self, dir: &Dir, print: bool) -> bool {
-		self.make_move(&dir);
+	pub fn enter_move(&mut self, dir: Dir, print: bool) -> bool {
+		self.make_move(dir);
```

Comment:
it's a simple enum, so like `u8`, it's better to pass it by value, not reference ;)